jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/AsyncEventNodeInstance.java

```diff
@@ -93,8 +93,10 @@ public Node getNode() {
     @Override
     protected NodeInstance getFrom() {
         AsyncEventNode node = (AsyncEventNode) getNode();
+        int level = getLevel();
         org.jbpm.workflow.instance.NodeInstance instance = ((org.jbpm.workflow.instance.NodeInstanceContainer) getNodeInstanceContainer()).getNodeInstance(node.getPreviousNode());
         ((org.jbpm.workflow.instance.NodeInstanceContainer) getNodeInstanceContainer()).removeNodeInstance(instance);
+        ((org.jbpm.workflow.instance.impl.NodeInstanceImpl) instance).setLevel(level);
```

Comment:
Does it have to be split in 2 commands for some reason?

Reply:
I don't understand what you mean :/ can you be more specific ?. There is no way to change removeInstance and not affect other parts of the code.

Reply:
Sorry, GitHub cut out one line, so the context is not full :)
I think you can do just
`((org.jbpm.workflow.instance.impl.NodeInstanceImpl) instance).setLevel(getLevel());`
I am not saying it's wrong, it is just a matter of preference, but I was curious if there is any side-effect because of which it has to be split into 2 separate statements.

Reply:
when you create the instance you change the level already. so it does not mean the same AFAIK